Adding pagination for GetBannedPersons

dessalines opened the pull request:
- Must come after https://github.com/LemmyNet/lemmy/pull/5424
- Fixes https://github.com/LemmyNet/lemmy/issues/2847

Nutomic commented:
With https://github.com/LemmyNet/lemmy/pull/5364 I changed it so only locally banned users are listed which fixes https://github.com/LemmyNet/lemmy/issues/2961. Seems like that got lost. 

While youre at it you can add a new endpoint  ` /api/v4/admin/list_users`. There it would be useful to include the email (but not all of LocalUser because its otherwise just various settings).

Does this also return pagination data for admins? Because that wouldnt make sense.

dessalines commented:
> With https://github.com/LemmyNet/lemmy/pull/5364 I changed it so only locally banned users

Fixed

> While youre at it you can add a new endpoint  /api/v4/admin/list_users. There it would be useful to include the email (but not all of LocalUser because its otherwise just various settings).

This one I'm still not sure about. The only two person type API selects, are listing banned users, and listing admins. And since admins already come back with GetSite, it doesn't really make sense to make that more generic. 

> Does this also return pagination data for admins? Because that wouldnt make sense.

Admins only come back with GetSite, with no pagination data. I'll add a *ignore_limits* to the `PersonQuery` to make sure that one isn't paged.

Nutomic commented on crates/db_views/src/person/person_view.rs:
```diff
@@ -37,33 +65,63 @@ impl PersonView {
+      query = query.filter(local_user::admin);
+    }
+
+    if !self.ignore_page_limits.unwrap_or(false) {
```
You can remove this param and ignore page limits whenever `admins_only == true`

dessalines replied:
Good  call
